src/resources/parser/glb-parser.js

```diff
@@ -253,6 +207,142 @@ Object.assign(pc, function () {
+                                (buffer.byteOffset),
+                        stride: bufferView.hasOwnProperty('byteStride') ? bufferView.byteStride : size,
+                        count: accessor.count
+                    };
+                }
+            }
+        }
+
+        // get position attribute
+        var positionDesc = sourceDesc[pc.SEMANTIC_POSITION];
+        var numVertices = positionDesc.count;
+
+        // generate normals if they're missing (this should probably be a user option)
+        if (!sourceDesc.hasOwnProperty(pc.SEMANTIC_NORMAL)) {
+            var positions = getAccessorData(accessors[attributes.POSITION], bufferViews, buffers);
+            generateNormals(sourceDesc, vertexDesc, positions, numVertices, indices);
+        }
+
+        return createVertexBufferInternal(device, numVertices, vertexDesc, positionDesc, sourceDesc);
+    };
+
+    var createVertexBufferDraco = function (device, outputGeometry, extDraco, decoder, decoderModule, semanticMap, indices) {
+
+        var numPoints = outputGeometry.num_points();
+
+        // helper function to decode data stream with id to TypedArray of appropriate type
+        var extractDracoAttributeInfo = function (uniqueId, storageType, componentSizeInBytes, normalize) {
+            var attribute = decoder.GetAttributeByUniqueId(outputGeometry, uniqueId);
+            var numValues = numPoints * attribute.num_components();
+            componentSizeInBytes = normalize ? 4 : componentSizeInBytes;   // if normalized, always decompress to float32 buffer
+            var dataSize = numValues * componentSizeInBytes;
+            var ptr = decoderModule._malloc( dataSize );
+            var values, k;
+
+            switch (storageType) {
+                case pc.TYPE_FLOAT32:
+                    decoder.GetAttributeDataArrayForAllPoints(outputGeometry, attribute, decoderModule.DT_FLOAT32, dataSize, ptr);
+                    values = new Float32Array(decoderModule.HEAPF32.buffer, ptr, numValues).slice();
+                    break;
+
+                case pc.TYPE_UINT8:
+                    if (normalize) {
+                        decoder.GetAttributeDataArrayForAllPoints(outputGeometry, attribute, decoderModule.DT_FLOAT32, dataSize, ptr);
+                        valuesFloat32 = new Float32Array(decoderModule.HEAPF32.buffer, ptr, numValues).slice();
+
+                        values = new Uint8ClampedArray(numValues);
+                        for (k = 0; k < numValues; k++)
+                            values[k] = valuesFloat32[k] * 255;
+                    } else {
+                        decoder.GetAttributeDataArrayForAllPoints(outputGeometry, attribute, decoderModule.DT_UINT8, dataSize, ptr);
+                        values = new Uint8Array(decoderModule.HEAPU8.buffer, ptr, numValues).slice();
+                    }
+                    break;
+
+                default:
+                    console.error("Element type not implemented: " + storageType);
```

Comment:
I'd wrap this (and any console logging) with an #ifdef DEBUG

Reply:
done